raiden_contracts/contract_manager.py

```diff
@@ -174,7 +174,8 @@ def merge_deployment_data(dict1: DeployedContracts, dict2: DeployedContracts) ->
     if not dict2:
         return dict1
     common_contracts: Dict[str, DeployedContract] = deepcopy(dict1["contracts"])
-    assert not common_contracts.keys() & dict2["contracts"].keys()
+    if common_contracts.keys() & dict2["contracts"].keys():
+        raise ValueError("dict1 and dict2 share some keys.")
```

Comment:
Maybe some explanation as why they sharing some key is bad would be useful

Reply:
I made a more detailed explanation here.